src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java

```diff
@@ -0,0 +1,94 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.imports.RedundantImportCheck;
+
+public class XpathRegressionRedundantImportTest extends AbstractXpathTestSupport {
+
+    private final String checkName = RedundantImportCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne(List<String> expectedXpathQueries) throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionRedundantImport1.java"));
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(RedundantImportCheck.class);
+        final String[] expectedViolation = {
+                "3:104: " + getCheckMessage(RedundantImportCheck.class,
+                        RedundantImportCheck.MSG_KEY, "org.checkstyle.suppressionxpathfilter.redundantimport.SuppressionXpathRegressionRedundantImport1"),
```

Comment:
`RedundantImportCheck` doesn't have any variable named `MSG_KEY`, it has `MSG_LANG`, `MSG_SAME`, `MSG_DUPLICATE`. A message key points to the complete error message in `.properties` file, example-
```
public static final String MSG_LANG = "import.lang";
```
`import.lang` is pointing to complete message in `.properties` file
https://github.com/checkstyle/checkstyle/blob/b68f16193f4db8b964797ae3a708aac29b296365/src/main/resources/com/puppycrawl/tools/checkstyle/checks/imports/messages.properties#L15

This key-value pair helps us show error messages in multiple languages-
https://github.com/checkstyle/checkstyle/blob/b68f16193f4db8b964797ae3a708aac29b296365/src/main/resources/com/puppycrawl/tools/checkstyle/checks/imports/messages_fr.properties#L15

Reply:
Done

Reply:
Resolved